gslab_scons/_release_tools.py

```diff
@@ -174,79 +169,54 @@ def upload_asset(github_token, org, repo, release_id, file_name,
     return r.content
 
 
-def up_to_date(mode = 'scons', directory = '.', scons_local_path = None):
+def scons_up_to_date(scons_local_path):
     '''
-    If mode = scons, this function checks whether the targets of a 
-    directory run using SCons are up to date. 
-    If mode = git, it checks whether the directory's working tree is clean.
+    Execute a SCons dry run and check that all the build steps are already up-to-date.
     '''
-    if mode not in ['scons', 'git']:
-        raise ReleaseError("up_to_date()'s mode argument must be "
-                           "'scons' or 'git'")
-
-    original_directory = os.getcwd()
-    os.chdir(directory)
-
-    if mode == 'scons':
-        # If mode = scons, conduct a dry run to check whether 
-        # all targets are up-to-date
-        if scons_local_path is not None:
-            scons_installation = 'python %s ' % os.path.join(directory, 
-                                                             scons_local_path)
-        else:
-            scons_installation = 'scons '
-        command = scons_installation + directory + ' --dry-run'
+    # Determine whether we're in an SCons directory.
+    if not os.path.isfile('SConstruct'):
+        raise ReleaseError('scons_up_to_date must be run on an SCons directory.')
+    # Execute the dry run and log its output.
+    if scons_local_path is not None:
+        scons_installation = 'python ' + scons_local_path
     else:
-        # If mode = git, check whether the working tree is clean.
-        command = 'git status'
-    
-    logpath = '.temp_log_up_to_date'
-    
+        scons_installation = 'scons '
+    command = scons_installation + ' --dry-run'
+    log = execute_up_to_date(command)
+    # Determine if SCons could start to run based on SCons logging.
+    if not check_list_for_regex('scons: Reading SConscript files', log):
+        raise ReleaseError('scons_up_to_date must be able to begin SCons build process.')
+    # Look for a line stating that the directory is up to date.
+    result = check_list_for_regex('is up to date\.$', log)
+    return result
+
+
+def git_up_to_date():
+    '''
+    Execute a git status and check that all changes are committed.
+    '''
+    # If mode = git, check whether the working tree is clean.
+    command = 'git status'
+    log = execute_up_to_date(command)
+    # Determine whether we're in a git repository based on git logging.
+    if check_list_for_regex('Not a git repository', log):
+        raise ReleaseError('git_up_to_date must be run on a git repository.')
+    result = check_list_for_regex('nothing to commit, working tree clean', output)
```

Comment:
@arosenbe would you be able to take a closer look at this file? Here the `output` variable isn't defined. You should probably check the `master` version to see how this variable was defined originally.

Reply:
Ah, sorry again. I changed the name output to log everywhere, except in apparently one location.